biodatasets/scai_chemical/scai_chemical.py

```diff
@@ -0,0 +1,256 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A dataset loader for the SCAI Chemical dataset.
+
+SCAI Chemical is a corpus of MEDLINE abstracts that has been annotated
+to give an overview of the different chemical name classes
+found in MEDLINE text.
+"""
+
+import gzip
+from typing import List, Tuple, Dict
+
+import datasets
+from utils import schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_CITATION = """\
+@inproceedings{kolarik:lrec-ws08,
+  author    = {Kol{\'a}{\vr}ik, Corinna and Klinger, Roman and Friedrich, Christoph M and Hofmann-Apitius, Martin and Fluck, Juliane},
+  title     = {Chemical Names: {T}erminological Resources and Corpora Annotation},
+  booktitle = {LREC Workshop on Building and Evaluating Resources for Biomedical Text Mining},
+  year      = {2008},
+}
+"""
+
+_DATASETNAME = "scai_chemical"
+
+_DESCRIPTION = """\
+SCAI Chemical is a corpus of MEDLINE abstracts that has been annotated
+to give an overview of the different chemical name classes
+found in MEDLINE text.
+"""
+
+_HOMEPAGE = "https://www.scai.fraunhofer.de/en/business-research-areas/bioinformatics/downloads/corpora-for-chemical-entity-recognition.html"
+
+_LICENSE = ""
+
+_URLS = {
+    _DATASETNAME: "https://www.scai.fraunhofer.de/content/dam/scai/de/downloads/bioinformatik/Corpora-for-Chemical-Entity-Recognition/chemicals-test-corpus-27-04-2009-v3_iob.gz",
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
+
+_SOURCE_VERSION = "3.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+
+class ScaiDiseaseDataset(datasets.GeneratorBasedBuilder):
```

Comment:
Should this be the class name?

```suggestion
class ScaiChemicalDataset(datasets.GeneratorBasedBuilder):
```